VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentCalendarButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { AppointmentAttributes, AppointmentLocation } from 'api/types'
+import { Box, LinkWithAnalytics } from 'components'
+import { Events } from 'constants/analytics'
+import { NAMESPACE } from 'constants/namespaces'
+import { logAnalyticsEvent } from 'utils/analytics'
+import {
+  AppointmentDetailsScreenType,
+  AppointmentDetailsSubType,
+  AppointmentDetailsSubTypeConstants,
+  AppointmentDetailsTypeConstants,
+  getAppointmentAnalyticsDays,
+  getAppointmentAnalyticsStatus,
+} from 'utils/appointments'
+import { getEpochSecondsOfDate } from 'utils/formattingUtils'
+import { useTheme } from 'utils/hooks'
+import { isIOS } from 'utils/platform'
+import { addToCalendar, checkCalendarPermission, requestCalendarPermission } from 'utils/rnCalendar'
+
+type AppointmentCalendarButtonProps = {
+  appointmentID: string
+  attributes: AppointmentAttributes
+  subType: AppointmentDetailsSubType
+  type: AppointmentDetailsScreenType
+}
+
+function AppointmentCalendarButton({ appointmentID, attributes, subType, type }: AppointmentCalendarButtonProps) {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const theme = useTheme()
+  const { location, minutesDuration, startDateUtc } = attributes || ({} as AppointmentAttributes)
+  const { address, lat, long, name } = location || ({} as AppointmentLocation)
+
+  const getLocation = (): string => {
+    if (isIOS() && lat && long) {
+      return name || ''
+    } else if (address?.street && address?.city && address?.state && address?.zipCode) {
+      return `${address.street} ${address.city}, ${address.state} ${address.zipCode}`
+    } else {
+      return name || ''
+    }
+  }
+
+  const getCalendarTitle = () => {
+    switch (type) {
+      case AppointmentDetailsTypeConstants.InPersonVA:
+        return t('upcomingAppointments.vaAppointment')
+      default:
+        return ''
+    }
+    //[AppointmentTypeConstants.COMMUNITY_CARE]: 'upcomingAppointments.communityCare',
```

Comment:
I left this comment here just so we know what to supply to the add calendar press for the title of the meeting since that didn't change. This will be removed once all the tickets are done so I think it is fine leaving this here